internal/p2p/transport_test.go

```diff
@@ -270,7 +294,10 @@ func TestConnection_Handshake(t *testing.T) {
 				assert.Equal(t, aInfo, peerInfo)
 				assert.Equal(t, aKey.PubKey(), peerKey)
 			}
-			errCh <- err
+			select {
+			case errCh <- err:
+			case <-ctx.Done():
```

Comment:
Do we want to return in this case? Previously we would have stalled indefinitely.

Reply:
Yes, but then we will orphan the reader at the other end, so I'm not sure we've solved that problem (I guess we've cut it in half, perhaps 🙂). Perhaps just ensure we close errCh before returning.

Reply:
well. I think the test passes and the channel is buffered and frankly I'm terrified, and also pretty sure that we're going to delete it when we do the p2p refactor so I'm inclined to leave it.